sdk/src/clients/consumer.rs

```diff
@@ -0,0 +1,743 @@
+use crate::client::Client;
+use crate::consumer::{Consumer, ConsumerKind};
+use crate::diagnostic::DiagnosticEvent;
+use crate::error::IggyError;
+use crate::identifier::{IdKind, Identifier};
+use crate::locking::{IggySharedMut, IggySharedMutFn};
+use crate::messages::poll_messages::{PollingKind, PollingStrategy};
+use crate::models::messages::{PolledMessage, PolledMessages};
+use crate::utils::crypto::Encryptor;
+use crate::utils::duration::{IggyDuration, SEC_IN_MICRO};
+use crate::utils::timestamp::IggyTimestamp;
+use bytes::Bytes;
+use futures::Stream;
+use futures_util::FutureExt;
+use std::collections::VecDeque;
+use std::future::Future;
+use std::pin::Pin;
+use std::sync::atomic::{AtomicBool, AtomicU32, AtomicU64};
+use std::sync::Arc;
+use std::task::{Context, Poll};
+use std::time::Duration;
+use tokio::time::sleep;
+use tracing::{error, info, trace, warn};
+
+const ORDERING: std::sync::atomic::Ordering = std::sync::atomic::Ordering::SeqCst;
+type PollMessagesFuture = Pin<Box<dyn Future<Output = Result<PolledMessages, IggyError>>>>;
+
+/// The auto-commit configuration for storing the offset on the server.
+#[derive(Debug, PartialEq, Copy, Clone)]
+pub enum AutoCommit {
```

Comment:
maybe ever N'th message mode would be useful ?

Reply:
Done